src/Symfony/Bundle/FrameworkBundle/Command/ServerLogCommand.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\VarDumper\Cloner\Data;
+use Symfony\Component\VarDumper\Dumper\CliDumper;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class ServerLogCommand extends Command
+{
+    private $dumper;
+    private $output;
+
+    private static $levelColorMap = [
+        100 => 'fg=white',
+        200 => 'fg=green',
+        250 => 'fg=blue',
+        300 => 'fg=cyan',
+        400 => 'fg=yellow',
+        500 => 'fg=red',
+        550 => 'fg=red',
+        600 => 'fg=white;bg=red',
+    ];
+
+    private static $bgColor = [
+        'black',
+        'red',
+        'green',
+        'yellow',
+        'blue',
+        'magenta',
+        'cyan',
+        'white',
+    ];
+
+    protected function configure()
+    {
+        $this
+            ->setName('server:log')
+            ->setDescription('Display log')
+            ->addOption('address', null, InputOption::VALUE_REQUIRED, 'The server address', '0:9911')
+            ->addOption('date-format', null, InputOption::VALUE_REQUIRED, 'The date format', 'H:i:s')
+            ->addOption('format', null, InputOption::VALUE_REQUIRED, 'The line format', '%s <%s>%-9s</> <options=bold>%-8.8s</> %s')
+            ->addOption('display-context', null, InputOption::VALUE_NONE, 'Display the context')
+            ->addOption('display-extra', null, InputOption::VALUE_NONE, 'Display the extra')
+            ->addOption('filter', null, InputOption::VALUE_REQUIRED, 'An expression to filter log. Example: record[\'level\'] > 200 or record[\'channel\'] in [\'app\', \'doctrine\']"')
+        ;
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->dumper = new CliDumper();
+        $this->dumper->setOutput($this->output = fopen('php://memory', 'r+b'));
+
+        $this->el = new ExpressionLanguage();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $logs = $this->getLogs($input);
+
+        $filter = $input->getOption('filter');
+
+        foreach ($logs as $message) {
+            $record = unserialize(base64_decode($message));
+
+            // Impossible to decode the message, give up.
+            if (false === $record) {
+                continue;
+            }
+
+            if ($filter && !$this->el->evaluate($filter, ['record' => $record])) {
```

Comment:
What about evaluate directly ``$record`` with ``->evaluate($filter, $record)``? ``filter`` option becomes easier to use: from ``--filter "record['level'] > 200"`` to ``--filter "level > 200"``